pkg/model/resources/nodeup.go

```diff
@@ -76,29 +92,28 @@ download-or-bust() {
 
   while true; do
     for url in "${urls[@]}"; do
-      commands=(
-        "curl -f --compressed -Lo "${file}" --connect-timeout 20 --retry 6 --retry-delay 10"
-        "wget --compression=auto -O "${file}" --connect-timeout=20 --tries=6 --wait=10"
-        "curl -f -Lo "${file}" --connect-timeout 20 --retry 6 --retry-delay 10"
-        "wget -O "${file}" --connect-timeout=20 --tries=6 --wait=10"
-      )
-      for cmd in "${commands[@]}"; do
-        echo "Attempting download with: ${cmd} {url}"
-        if ! (${cmd} "${url}"); then
-          echo "== Download failed with ${cmd} =="
-          continue
-        fi
-        if ! validate-hash "${file}" "${hash}"; then
-          echo "== Hash validation of ${url} failed. Retrying. =="
-          rm -f "${file}"
-        else
-          echo "== Downloaded ${url} (SHA256 = ${hash}) =="
-          return 0
-        fi
-      done
-    done
-
-    echo "All downloads failed; sleeping before retrying"
+	  local access_token=""
+	  if [[ "$GCE_ACCESS_TOKEN_URL" != "" ]]; then
+		# The token always begins with 'ya29' and continues until the next double-quote.
+		# We would normally use jq to be more tolerant of format changes, 
+		# but it's not available in flatcar during startup (depends on docker, which isn't up yet.)
```

Comment:
TIL that the GCP metadata service supports an alt=text parameter, and we've even documented it :-)   https://cloud.google.com/compute/docs/metadata/querying-metadata#formatting_query_output

That might be easier to parse, though I don't see this as a blocker.